enhancements/sha-bundle-digest.md

```diff
@@ -0,0 +1,136 @@
+  digests. These variables should be set in `config/manager/manager.yaml` as suggested in
+  [https://master.sdk.operatorframework.io/docs/best-practices/common-recommendation/#other-common-suggestions](https://master.sdk.operatorframework.io/docs/best-practices/common-recommendation/#other-common-suggestions).
+- Update CSV relatedImages with resolved image tags to digests
+
+```yaml
+# Image w/ traditional tags
+- name: RELATED_IMAGE_che_server_secure_exposer_jwt_proxy_image
+  value: quay.io/eclipse/che-jwtproxy:0.10.0
+
+# Image w/ SHA based digest
+- name: RELATED_IMAGE_single_host_gateway
+  value: quay.io/eclipse/che--traefik:v2.5.0-eb30f9f09a65cee1fab5ef9c64cb4ec91b800dc3fdd738d62a9d4334f0114683
+```
+
+### Risks and Mitigations
+
+Since the feature is opt-in, it carries the risk of the end-user enabling it without a correct understanding of the
+feature and its implications. This can be mitigated with clear documentation both on the website and in the CLI help
+section.
+
+## Design Details
+
+- As of this EP, the new `--use-image-digests` is the only flag to pass to the `operator-sdk generate bundle` command.
+  In order to allow users full control over the command's flags, the `Makefile` should use an intermediary variable to
+  hold the flags that a user can overwrite entirely. For example:
+
+```Makefile
+GENERATE_BUNDLE_FLAGS ?= ""
+USE_IMAGE_DIGESTS ?= false
+ifeq ($(GENERATE_BUNDLE_FLAGS),"")
+    ifeq($(USE_IMAGE_DIGESTS),true)
+        GENERATE_BUNDLE_FLAGS += "--use-image-digests"
+    endif
+endif
+bundle:
+    operator-sdk generate bundle $(GENERATE_BUNDLE_FLAGS)
+```
+
+- The `RELATED_IMAGE_*` variables should be defined in the `manager.yaml` file as outlined
+  [here](https://master.sdk.operatorframework.io/docs/best-practices/common-recommendation/#other-common-suggestions).
+- In order to resolve digests from tags, the CLI will use the `imageresolver` package in
+  [operator-manifest-tools](https://github.com/operator-framework/operator-manifest-tools).
+- The output cluster CSV will have the `relatedImage` annotation that is explained in the
+  [OpenShift Documentation](https://docs.openshift.com/container-platform/4.9/operators/operator_sdk/osdk-generating-csvs.html#olm-enabling-operator-for-restricted-network_osdk-generating-csvs).
+
+### Test Plan
+
+The resulting artifact of this EP will be a cluster CSV with image digests instead of tags in the image definitions as
+well as digest annotations. This feature will need an integration test that veifiesthat the knobs we provide to enable
+and configure image digests generates an expected CSV. The expected CSV will be based off of the existing memcached CSV,
+located
+[here](https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v3/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml)
+with the `spec.relatedImages` section from the proposal section added to it.
```

Comment:
```suggestion
The resulting artifact of this EP will be a CSV with image digests instead of tags in the image definitions as well as
digest annotations. This feature will need an integration test that verifies that the knobs we provide to enable and
configure image digests generate an expected CSV. The expected CSV will be based on the example memcached CSV with the
proposed changes (see [Appendix B](#appendix-b)).
```